src/components/input/SelectNext.tsx

```diff
@@ -90,13 +91,17 @@ function SelectMain<T>({
   elementRef,
 }: SelectProps<T>) {
   const [listboxOpen, setListboxOpen] = useState(false);
-  const [shouldListboxDropUp, setShouldListboxDropUp] = useState(false);
   const closeListbox = useCallback(() => setListboxOpen(false), []);
   const wrapperRef = useRef<HTMLDivElement | null>(null);
   const listboxRef = useRef<HTMLDivElement | null>(null);
   const listboxId = useId();
   const buttonRef = useSyncedRef(elementRef);
   const buttonId = useId();
+  const shouldListboxDropUp = useShouldBePositionedAbove(
```

Comment:
If this is an internal helper not used by other components, it could just go inside the same `SelectNext` module. Then it doesn't have to pretend to be generic.

Reply:
That's probably the best approach at this point. We can keep the naming close to select/listbox and think on better names and extracting it if the need apperas.